src/GSF.CarbonAware/test/EmissionsHandlerTests.cs

```diff
@@ -15,55 +18,153 @@ namespace GSF.CarbonAware.Tests;
-    /// Tests that successfull call to the aggregator with any data returned results in expected format.
+    /// GetBestEmissionsData: Tests empty or null location arrays throw ArgumentException.
     /// </summary>
-    [TestCase("Sydney", "2022-03-07T01:00:00", "2022-03-07T03:30:00", TestName = "GetAverageCarbonIntensity calls aggregator successfully")]
-    public async Task GetAverageCarbonIntensity_SuccessfulCallReturnsOk(string location, DateTimeOffset start, DateTimeOffset end)
+    [TestCase(new object?[] { null, null }, TestName = "GetBestEmissions, array of nulls, throws: simulates 'location=&location=' empty value input")]
+    [TestCase(new object?[] { null, }, TestName = "GetBestEmissions, array of nulls, throws: simulates 'location=' empty value input")]
+    [TestCase(new object?[] { }, TestName = "GetBestEmissions, empty array, throws: simulates no 'location' query string")]
+    public void GetBestEmissionsData_NoLocations_ThrowsException(params string[] locations)
     {
         // Arrange
-        double data = 0.7;
-        var emissionsHandler = new EmissionsHandler(Logger!.Object, CreateEmissionsAggregatorWithAverageCI(data).Object);
+        var emissionsHandler = new EmissionsHandler(Logger!.Object, CreateEmissionsAggregatorWithBestEmissionsData(EmptyTestData).Object);
+
+        //Act/Assert
+        Assert.ThrowsAsync<ArgumentException>(async () => await emissionsHandler.GetBestEmissionsDataAsync(locations));
+    }
 
-        var parametersDTO = new CarbonAwareParametersBaseDTO
+    /// <summary>
+    /// GetBestEmissionsData: Tests that successful emissions call to aggregator with varied location input returns expected data.
+    /// </summary>
+    [TestCase(new object?[] { null, "Sydney" }, TestName = "GetBestEmissions, successful: simulates 'location=&location=Sydney'")]
+    [TestCase(new object?[] { "Sydney", null }, TestName = "GetBestEmissions, successful: simulates 'location=Sydney&location='")]
+    [TestCase(new object?[] { "Sydney", "Melbourne" }, TestName = "GetBestEmissions, successful: simulates 'location=Sydney&location=Melbourne'")]
+    [TestCase(new object?[] { "Sydney" }, TestName = "GetBestEmissions, successful: simulates 'location=Sydney'")]
+    public async Task GetBestEmissionsData_VariedLocationsInput_SuccessfulCall_ReturnsOk(params string[] locations)
+    {
+        // Arrange
+        var data = new EmissionsData[]
         {
-            SingleLocation = location,
-            Start = start,
-            End = end
+            new EmissionsData()
+            {
+                Location = "Sydney",
+                Rating = 0.9,
+                Time = DateTime.Now
+            }
         };
+        var emissionsHandler = new EmissionsHandler(Logger!.Object,CreateEmissionsAggregatorWithBestEmissionsData(data).Object);
+
+        // Act
+        var result = await emissionsHandler.GetBestEmissionsDataAsync(locations);
+
+        // Assert
+        Assert.IsNotEmpty(result);
```

Comment:
It would be good to be in-sync with all the versioning used for `NUnit, Microsoft.NET.Test.Sdk, NUnit3TestAdapter` accross the project (check for instance EM project). Also include `<PackageReference Include="NUnit.Analyzers" Version="3.3.0" />`.
At build time a `warning` will show up using for instance `Assert.IsNotEmpty`. It would make this to change for `Assert.That(...)`